File: frontend/src/service/transactionService.ts
import type {
  TransactionResponse,
  TransactionRequest,
  PageResponse,
} from "../../src/types/transaction";
import api from "../config/axios";

export const getTransactions = async (params: any) => {
  const response = await api.get<any>("/transactions/admin", {
    params,
  });

  return response.data.result as PageResponse<TransactionResponse>;
};

export const getRentalAreaTransactions = async (
  rentalAreaId: string,
  params: any,
) => {
  const response = await api.get<any>(
    `/transactions/rental-area/${rentalAreaId}`,
    {
      params,
    },
  );

  return response.data.result as PageResponse<TransactionResponse>;
};

export const createTransaction = async (data: TransactionRequest) => {
  const response = await api.post<any>("/transactions", data);
  return response.data.result as TransactionResponse;
};

export const updateTransaction = async (
  id: string,
  data: TransactionRequest,
) => {
  const response = await api.put<any>(`/transactions/${id}`, data);
  return response.data.result as TransactionResponse;
};

File: backend/src/main/java/org/sport/backend/service/TransactionService.java
package org.sport.backend.service;

import org.sport.backend.constant.TransactionType;
import org.sport.backend.dto.base.PageResponse;
import org.sport.backend.dto.request.transaction.TransactionRequest;
import org.sport.backend.dto.response.transaction.TransactionResponse;
import org.sport.backend.dto.response.transaction.TransactionSummaryResponse;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

public interface TransactionService {
    PageResponse<TransactionResponse> getRentalAreaTransactions(
            UUID rentalAreaId,
            int page,
            int size,
            String keyword,
            TransactionType type,
            LocalDateTime startDate,
            LocalDateTime endDate
    );
    PageResponse<TransactionResponse> getTransactions(
            int page,
            int size,
            String keyword,
            TransactionType type,
            LocalDateTime startDate,
            LocalDateTime endDate
    );

    PageResponse<TransactionResponse> getOwnerTransactions(
            UUID ownerId,
            int page,
            int size,
            TransactionType type
    );

    TransactionResponse createTransaction(TransactionRequest request);

    TransactionResponse updateTransaction(UUID id, TransactionRequest request);

    TransactionSummaryResponse getRentalAreaTransactionSummary(UUID rentalAreaId, LocalDate startDate, LocalDate endDate);
}
